RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,301 @@
+The shared identification keys ensure that all members of that committee are able to receive and decrypt data messages that were sent to the committee.
+
+#### Maintaining connections with peers
+
+CCs and CNs establish and maintain connections with peers differently.
+CCs only create a few short-lived ad hoc channels and CNs create and maintain long-lived channels with a number of peers.
+
+If a CC is unaware of a destination CNs or CCs online communication address then the address first needs to be obtained using a discovery request.
+When a CC already know the communication address of the CC or CN that he wants to communicate with, then a direct P2P channel can be established between the two peers for the duration of the communication task.
+The communication channel can then be closed as soon as the communication task has been completed.
+
+CNs consisting of VNs and BNs typically attempt to maintain communication channels with a large number of peers.
+The distribution of peers (VNs vs BNs) that a single CN keeps communication channels open with can change depending on the type of node.
+A CN that is also a BN should maintain more peer connections with other BNs, but should also have some connections with other VNs.
+
+Having some connections with VNs are important as BNs have derived node ids and not registered node ids such as VNs, making it possible for the CN to be separated from the main network and become victim to an eclipse attack.
+Having some connections with VNs will make it more difficult to separate the CN from the network and will ensure successful propagation of transactions and completed blocks from that CN. 
+
+A CN that is also a VN should maintain more peer connections with other VNs, but also have some connections with BNs.
+CNs that are part of Validator Node committees should attempt to maintain permanent connections with the other members of the committee to ensure that quick consensus can be achieved.
+
+To maintain connections with peers, the following process can be performed.
+Once some peers have been found using discovery requests, resulting in them successfully being added to the local routing table.
+The CN can decide how the peer connections should be selected from the routing table by either:
+ - manually selecting a subset,
+ - automatically selecting a random subset or
+ - selecting a subset of neighbouring nodes with similar node ids. 
+
+ Maintaining communication channels are important and the following process can be followed in an attempt to keep peer connections alive:
+For an existing peer connection.
+When more than 30 minutes have passed since the last communication with that peer, then a heartbeat message should be sent to the peer in an attempt to keep the connection with the peer alive.
+If that specific peer connection is not important then a new peer can be selected from the local routing table.
```

Comment:
> is not important 
That's a very vague descriptor. Is it possible to be more specific here?

Reply:
I revised the sentence to be more descriptive: "If the peer connection was not established for a specific purpose, such as with the connections between committee members, then a new replacement peer can be selected from the local routing table."

Reply:
👍